pkg/asset/agent/mirror/registriesconf.go

```diff
@@ -208,6 +214,23 @@ func (i *RegistriesConf) finish() error {
 	return nil
 }
 
+func (i *RegistriesConf) validateReleaseImageIsSameInRegistriesConf(source string) error {
+
+	releaseImage := &releaseimage.Image{}
+	releaseImage.Generate(asset.Parents{})
```

Comment:
Is there a reason we can't depend on this asset and get it from the store instead of generating a new one?

Reply:
Not sure. I referred https://github.com/openshift/installer/blob/master/pkg/asset/agent/manifests/clusterimageset.go#L146-L147. I will look into getting it from store ( and what that even means :laughing: ) Probable we shoudl also change it in clusterimageset then.

Reply:
@zaneb I am able to use the get the Image asset from the store in case of `Generate()` but in case of `Load()` I think, the asset needs to be generated.

Reply:
If we depend on it (i.e. it is returned in the list from `Dependencies()`) then it will get generated.

Reply:
Oh, I see what you're saying. No list of parent dependencies get passed to `Load()`, so we don't have a way of fetching it. Yeah, this is unavoidable then.